GPU/GPUTracking/DataCompression/GPUTPCCompressionTrackModel.cxx

```diff
@@ -112,4 +112,923 @@ GPUd() int GPUTPCCompressionTrackModel::Mirror()
+    mC[2] = err2Z;
+    mNDF = -3;
+    return 0;
+  }
+  //propably not needed, definitely not needed for the first measurement
+  // constrain sin(phi)
+  if (mP[2] > GPUCA_MAX_SIN_PHI) {
+    mP[2] = GPUCA_MAX_SIN_PHI;
+  } else if (mP[2] < -GPUCA_MAX_SIN_PHI) {
+    mP[2] = -GPUCA_MAX_SIN_PHI;
+  }
+
+  const float d00 = mC[0], d01 = mC[1], d02 = mC[3], d03 = mC[6], d04 = mC[10];
+  const float d10 = mC[1], d11 = mC[2], d12 = mC[4], d13 = mC[7], d14 = mC[11];
+
+  const float z0 = y - mP[0];
+  const float z1 = z - mP[1];
+  float w0, w1, w2, chiY, chiZ;
+  if (mNDF <= 0) {
+    w0 = 1.f / (err2Y + d00);
+    w1 = 0;
+    w2 = 1.f / (err2Z + d11);
+    chiY = w0 * z0 * z0;
+    chiZ = w2 * z1 * z1;
+  } else {
+    w0 = d11 + err2Z, w1 = d10, w2 = d00 + err2Y;
+    { // Invert symmetric matrix
+      float det = w0 * w2 - w1 * w1;
+      if (CAMath::Abs(det) < 1.e-10f) {
+        return -1;
+      }
+      det = 1.f / det;
+      w0 = w0 * det;
+      w1 = -w1 * det;
+      w2 = w2 * det;
+    }
+    chiY = CAMath::Abs((w0 * z0 + w1 * z1) * z0);
+    chiZ = CAMath::Abs((w1 * z0 + w2 * z1) * z1);
+  }
+  float dChi2 = chiY + chiZ;
+  mChi2 += dChi2;
+  mNDF += 2;
+
+  if (mNDF <= 0) {
+    const float k00 = d00 * w0;
+    const float k20 = d02 * w0;
+    const float k40 = d04 * w0;
+    const float k11 = d11 * w2;
+    const float k31 = d13 * w2;
+    mP[0] += k00 * z0;
+    mP[1] += k11 * z1;
+    mP[2] += k20 * z0;
+    mP[3] += k31 * z1;
+    mP[4] += k40 * z0;
+
+    mC[0] -= k00 * d00;
+    mC[2] -= k11 * d11;
+    mC[3] -= k20 * d00;
+    mC[5] -= k20 * d02;
+    mC[7] -= k31 * d11;
+    mC[9] -= k31 * d13;
+    mC[10] -= k00 * d04;
+    mC[12] -= k40 * d02;
+    mC[14] -= k40 * d04;
+  } else {
+    const float k00 = d00 * w0 + d01 * w1;
+    const float k01 = d00 * w1 + d10 * w2;
+    const float k10 = d01 * w0 + d11 * w1;
+    const float k11 = d01 * w1 + d11 * w2;
+    const float k20 = d02 * w0 + d12 * w1;
+    const float k21 = d02 * w1 + d12 * w2;
+    const float k30 = d03 * w0 + d13 * w1;
+    const float k31 = d03 * w1 + d13 * w2;
+    const float k40 = d04 * w0 + d14 * w1;
+    const float k41 = d04 * w1 + d14 * w2;
+
+    mP[0] += k00 * z0 + k01 * z1;
+    mP[1] += k10 * z0 + k11 * z1;
+    mP[2] += k20 * z0 + k21 * z1;
+    mP[3] += k30 * z0 + k31 * z1;
+    mP[4] += k40 * z0 + k41 * z1;
+
+    mC[0] -= k00 * d00 + k01 * d10;
+
+    mC[2] -= k10 * d01 + k11 * d11;
+
+    mC[3] -= k20 * d00 + k21 * d10;
+    mC[5] -= k20 * d02 + k21 * d12;
+
+    mC[7] -= k30 * d01 + k31 * d11;
+    mC[9] -= k30 * d03 + k31 * d13;
+
+    mC[10] -= k40 * d00 + k41 * d10;
+    mC[12] -= k40 * d02 + k41 * d12;
+    mC[14] -= k40 * d04 + k41 * d14;
+
+    mC[1] -= k10 * d00 + k11 * d10;
+    mC[4] -= k20 * d01 + k21 * d11;
+    mC[6] -= k30 * d00 + k31 * d10;
+    mC[8] -= k30 * d02 + k31 * d12;
+    mC[11] -= k40 * d01 + k41 * d11;
+    mC[13] -= k40 * d03 + k41 * d13;
+  }
+  return 0;
+}
+
+GPUd() int GPUTPCCompressionTrackModel::Mirror()
```

Comment:
Not sure if this is used currently, might be that we break instead of following the loop. But please leve the functionality in in case we might want to store loopers later.

Reply:
The way I see it Mirror() is used for the merger track model. The slice tracker track model breaks for loopers. I will keep it for now.